newIDE/app/src/InstancesEditor/FullSizeInstancesEditorWithScrollbars.js

```diff
@@ -124,14 +84,51 @@ const FullSizeInstancesEditorWithScrollbars = (props: Props) => {
     hideScrollbarsAfterDelay();
   }, 500);
 
-  // If the mouse gets out of the detection zone, whilst not dragging, hide the scrollbars.
-  const mouseOutDetectionZoneHandler = React.useCallback(
-    (e: MouseEvent) => {
-      if (!isDragging.current) {
-        hideScrollbarsAfterDelay();
+  const showScrollbarsThrottled = throttle(
+    () => {
+      if (!showScrollbars.current) {
+        showScrollbars.current = true;
+        forceUpdate();
+      }
+      if (timeoutHidingScrollbarsId.current) {
+        clearTimeout(timeoutHidingScrollbarsId.current);
+        timeoutHidingScrollbarsId.current = null;
       }
     },
-    [hideScrollbarsAfterDelay]
+    1000,
+    { leading: true, trailing: false }
+  );
+
+  const hideScrollbarsAfterDelayThrottled = throttle(
+    hideScrollbarsAfterDelay,
+    1000,
+    { leading: true, trailing: false }
+  );
+
+  const onMouseMoveOverInstanceEditor = React.useCallback(
+    (event: MouseEvent) => {
+      if (!editorRef.current) {
+        return;
+      }
+
+      const shouldDisplayScrollBars = !canvasRectangle.containsPoint(
+        event.clientX,
+        event.clientY
+      );
+
+      if (shouldDisplayScrollBars) {
+        showScrollbarsThrottled();
+      } else {
+        if (!isDragging.current) {
+          hideScrollbarsAfterDelayThrottled();
```

Comment:
same here, if `showScrollbars.current` is already false, we probably don't need to call it? It means they're just moving around